packages/utils/src/liquidity.ts

```diff
@@ -0,0 +1,63 @@
+import type { EvmResource, SubstrateResource, Eip1193Provider } from '@buildwithsygma/core';
+import { Network, ResourceType } from '@buildwithsygma/core';
+import type { createEvmFungibleAssetTransfer } from '@buildwithsygma/evm';
+import { getEvmHandlerBalance } from '@buildwithsygma/evm';
+import type { createSubstrateFungibleAssetTransfer } from '@buildwithsygma/substrate';
+import { HttpProvider } from 'web3-providers-http';
+
+import { getSubstrateHandlerBalance } from './substrate/balances.js';
+
+export async function hasEnoughLiquidity(
+  transfer:
+    | Awaited<ReturnType<typeof createEvmFungibleAssetTransfer>>
+    | Awaited<ReturnType<typeof createSubstrateFungibleAssetTransfer>>,
+  destinationProviderUrl: string,
+): Promise<boolean> {
+  const destination = transfer.destination;
```

Comment:
can we use destructuring here instead of the getters? same question from the line below